core/src/main/scala/spinal/core/Data.scala

```diff
@@ -121,6 +121,11 @@ trait DataPrimitives[T <: Data]{
       }
     }
   }
+
+  def boot(that: T): T = {
```

Comment:
In practice how does it overlap with the ClockDomain resetKind BOOT ?

Reply:
I think the resetKind Boot can be removed, as it can be replaced by boot(that). I don t know which scenario will use it

Reply:
It often write as this:
 reg [7:0]    dataOut=8'h0;
It‘s OK written by:
initial begin
    dataOut = 8'h0;
  end
I‘m not sure how initial is implemented in VHDL

Reply:
should `reg [7:0] dataOut=8'h0;`  more  better than `initial begin  end` way

Reply:
> I think the resetKind Boot can be removed

It shouldn't be removed, main reason is that it allows to take a reset agnostic design, and port it on different tech, without having to change the way the user has to reset things, just by swaping the clock domain config.

> I‘m not sure how initial is implemented in VHDL

There is no initial begin equivalent in VHDL, only signal xxx : yyy  = zzz;

> should reg [7:0] dataOut=8'h0; more better than initial begin end way

I have no idea what is the best from a EDA tooling perspective.
The advantage of the initial begin, is that it allows more complex code. ex ; multi line partial assignement, for loop and stuff like this


One thing to notice, there is myReg.randBoot() ? (in order to avoid x.propagation on register without reset for instance), which will provide a random value.